plugins/serializers/prometheusremotewrite/prometheusremotewrite.go

```diff
@@ -195,7 +195,7 @@ func (s *Serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
 			// sample then we can skip over it.
 			m, ok := entries[metrickey]
 			if ok {
-				if metric.Time().Before(time.Unix(m.Samples[0].Timestamp, 0)) {
+				if metric.Time().Before(time.Unix(m.Samples[0].Timestamp/1000, 0)) {
```

Comment:
Doing this, you will loose the millisecond fraction of your timestamp. I don't know enough of the code to judge if this is a problem, but maybe
```suggestion
				if metric.Time().Before(time.Unix(0, m.Samples[0].Timestamp*1_000_000)) {
```
would be better?

Reply:
fixed